File: app/src/main/java/com/example/nasa/data/api/ApiKeyInterceptor.java
package com.example.nasa.data.api;

import androidx.annotation.NonNull;

import com.example.nasa.BuildConfig;

import java.io.IOException;

import okhttp3.HttpUrl;
import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class ApiKeyInterceptor implements Interceptor {
    @Override
    public @NonNull
    Response intercept(@NonNull Chain chain) throws IOException {
        Request request = chain.request();
        HttpUrl httpUrl = request
                .url()
                .newBuilder()
                .addQueryParameter("api_key", BuildConfig.API_KEY)
                .build();
        return chain.proceed(request
                .newBuilder()
                .url(httpUrl)
                .build());
    }
}

File: data/src/main/java/com/example/data/api/ApiKeyInterceptor.kt
package com.example.data.api

import com.example.data.BuildConfig
import okhttp3.Interceptor
import okhttp3.Response
import java.io.IOException

class ApiKeyInterceptor : Interceptor {

    companion object {
        private const val PROPERTY_API_KEY = "api_key"
    }

    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val request = chain.request()
        val httpUrl = request
                .url
                .newBuilder()
                .addQueryParameter(PROPERTY_API_KEY, BuildConfig.API_KEY)
                .build()
        return chain.proceed(request
                .newBuilder()
                .url(httpUrl)
                .build())
    }
}